sympy/logic/boolalg.py

```diff
@@ -2289,10 +2287,7 @@ def _compare_term(minterm, term):
     Return True if a binary term is satisfied by the given term. Used
     for recognizing prime implicants.
     """
-    for m, t in zip(minterm, term):
-        if t != 3 and m != t:
-            return False
-    return True
+    return all(t == 3 or minterm[n] == t for n, t in enumerate(term))
```

Comment:
Is it not better to use zip?

Reply:
No, as this enables early exit which zip doesn't. Clear speed-advantage here, statistically at least.

There is a handful of `any` and `all`-statements in the code that uses `zip` that one may want to consider at a later stage. One should probably check how large the iterators are though, for just a few items it may be slower I think, but for "enough" there will be a significant improvement.

Reply:
Why doesn't `zip` enable an early exit?

I was thinking of something like:
```
return all(t == 3 or t == mt for t, mt in zip(term, minterm)):
return all(t in (3, mt) for t, mt in zip(term, minterm)):
```

Reply:
You are right in that it is not the reason. However, my experiments show that it is faster. I think that the reason is that you can avoid fetching the minterm argument every now and then since it is enough to test if the term argument is 3 and only if not fetch the minterm argument. But I can try to recreate the timings if you think it is required.

Compare with the `_check_pairs` function which is faster without a zip, although that one also includes an enumerate. See https://github.com/sympy/sympy/commit/d3e5f14c5241b9da96560c4deffef9eaf4991228#diff-0fa274ccbfd01ea485764211bac9ecce2ac3819d1c110968d0b13c7da3bab6f0 for the earlier version.

(Now that it is a one-liner one can also replace it in the code, but that hold for all one-liners...)

Reply:
Nah, I may be confusing things. I realize that I change from a  loop with enumerate to the current zip-loop earlier... I'll try the three single-line cases and come back...

Reply:
Your first oneliner seems marginally faster then mine (times vary, maybe hard to say any is better). The second clearly slower than both.